scripts/check-hoisting.ts

```diff
@@ -71,54 +80,53 @@ const path = require('path');
         return issues;
     }
 
-    function versionOf(npmPackagePath) {
+    function versionOf(npmPackagePath: string): string {
         const packageJsonPath = path.join(npmPackagePath, 'package.json');
         if (fs.existsSync(packageJsonPath)) {
             return require(packageJsonPath).version || '';
         }
         return '';
     }
 
-    function warn(issues, extension, message) {
+    function warn(issues: DiagnosticMap, extension: string, message: string): void {
         return log(issues, extension, message, 'warn');
     }
 
-    function error(issues, extension, message) {
+    function error(issues: DiagnosticMap, extension: string, message: string): void {
         return log(issues, extension, message, 'error');
     }
 
-    function log(issues, extension, message, type) {
+    function log(issues: DiagnosticMap, extension: string, message: string, type: DiagnosticType): void {
         const key = `@theia/${extension}`;
         if (!issues.has(key)) {
             issues.set(key, []);
         }
         const severity = toSeverity(type);
-        issues.get(key).push({ severity, message });
-        return issues;
+        issues.get(key)!.push({ severity, message });
```

Comment:
The API was designed to return with the item you have logged and not `void`. Why? I do not mind if you change it but please be consistent.

Either add `return issues` here or do not return from `error` and `warn`.

Thanks!

Reply:
To be clear, it currently (before this patch) returns the map, not the just the new issue.

I changed it to return nothing, because nothing is using it.  I changed `error` and `warn` to remove the `return` statement.  It doesn't change anything, but I agree it's confusing to have it there.